### ISSUE TITLE
Use joinable! macro to join using non PK columns

### ISSUE THREAD

**Issue by PritiKumr** [diesel-rs/diesel#938]
Trying to use the `joinable!` macro for this issue on crates.io - https://github.com/rust-lang/crates.io/pull/740

We're trying to join the following tables - 

```
crate_downloads
(
  crate_id integer NOT NULL,
  downloads integer NOT NULL,
  date date NOT NULL,
  CONSTRAINT crate_downloads_pkey PRIMARY KEY (crate_id, date),
)
```

```
CREATE TABLE public.crate_owners
(
  crate_id integer NOT NULL,
  owner_id integer NOT NULL,
  owner_kind integer NOT NULL,
  CONSTRAINT crate_owners_pkey PRIMARY KEY (crate_id, owner_id, owner_kind),
)
```

Note - removed few columns for clarity.

I want to join these two tables on just the `crate_id` column. But this is part of a composite key on `crate_downloads`. So when we do  - 

```
joinable!(crate_owners -> crate_downloads (crate_id));
```

I get the error - 

```
error[E0271]: type mismatch resolving `<diesel::expression::nullable::Nullable<(schema::crate_downloads::columns::crate_id, schema::crate_downloads::columns::date)> as diesel::Expression>::SqlType == diesel::types::Nullable<diesel::types::Integer>`
   --> src/user/mod.rs:391:5
    |
391 |     joinable!(crate_owners -> crate_downloads (crate_id));
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected tuple, found struct `diesel::types::Integer`
    |
    = note: expected type `diesel::types::Nullable<(diesel::types::Integer, diesel::types::Date)>`
               found type `diesel::types::Nullable<diesel::types::Integer>`
    = note: this error originates in a macro outside of the current crate
```

Appreciate some guidance here. :)

**Comment by sgrif** [comment]
In theory we could allow this, but it would be more true to the associations API for this to go through the crates table, since that is the primary key both foreign keys are referencing. So this is at least partially a duplicate of the issues for adding through associations, but I'm going to leave this open since this is specific to declaring possible joins.

**Comment by steverob** [comment]
@sgrif it makes sense to go for a _through_ association for this. But, would it not be nice to let the developers choose the appropriate approach here and diesel giving them all options? :)

**Comment by sgrif** [comment]
This was fixed by 5cd4c0c2
